src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/helpers/ConfigDPHelper.ps1

```diff
@@ -107,26 +102,33 @@ function Invoke-RestMethodWithUriParameters {
     #     $uriParametersArray = $uriParameters.GetEnumerator() | ForEach-Object { "$($_.Key)=$($_.Value)" } | ForEach-Object { $_ -join '=' } | ForEach-Object { $_ -join '&' }
     # }
     Write-Debug "Issue REST request to ${uri} with method ${method} and headers ${headers} and body ${requestBody}"
-    $rsp = Invoke-RestMethod `
-        -Method $method `
-        -Uri $uri `
-        -Headers $headers `
-        -Body $requestBody `
-        -ContentType "application/json" `
-        -MaximumRetryCount $maximumRetryCount `
-        -RetryIntervalSec $retryintervalSec `
-        -StatusCodeVariable statusCode `
-        -Verbose:($PSCmdlet.MyInvocation.BoundParameters["Verbose"].IsPresent -eq $true) `
-        -Debug:($PSCmdlet.MyInvocation.BoundParameters["Debug"].IsPresent -eq $true)
+    try {
+        $rsp = Invoke-RestMethod `
```

Comment:
You can use `Invoke-WebRequest` as a nearly drop-in replacement which gives you the real status code, and works with PS 5.1

https://stackoverflow.com/a/64261797/6795825

Making up a 400 response if there's an error might be a pain for debugging

Reply:
I came across this method when resolving with unit tests errors, but further reading on the differences:
- `Invoke-RestMethod` is designed specifically for interacting with RESTful APIs. https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-restmethod?view=powershell-7.4
- `Invoke-WebRequest` is for general-purpose web request and will return raw content https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-webrequest?view=powershell-5.1

Given our usage, Invoke-RestMethod is appropriate and we don't need extra work to parses JSON to PSObject on response, I'm not sure if Invoke-WebRequest we used before and that's part of the JSON response fun we had weeks ago. You're right it could make the unit test easier, but could also break other bits.

Reply:
We chatted, it's probably not the best time to consider the replacement, given it may break other parts of the code.